### PR TITLE
Basic functioning comments section

### PR DESCRIPTION
**Pull request by tatiana-s**
The website now has a comment section with input form and persistent storage.

- Comments can  be added using a form on the main page instead of being hardcoded.
- They are then stored permanently in a Datastore database and can be loaded through queries.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -28,15 +33,33 @@
...
 public class DataServlet extends HttpServlet {
 
-  private final List<String> testComments = new ArrayList<String>(Arrays.asList( 
-    "This is an example comment.",
-    "Another example comment...",
-    "Wow! Yet another example comment?"));
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // load comments from database
```
Comment (csperl) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: When you add comments to code, please try to write full sentences, including an upper case character at the beginning and a period at the end :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -28,15 +33,33 @@
...
+    Query query = new Query("Comment");
```
Comment (csperl) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You use the constant "Comment" here and below. They need to be in sync. Best is to define them as private static final String objects in this class and add a comment what they are and what happens if one changes them.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -28,15 +33,33 @@
...
+    }
     response.setContentType("application/json;");
-    response.getWriter().println(convertToJson(testComments));
+    response.getWriter().println(convertToJson(comments));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // retrieve input
+    String comment = request.getParameter("comment-input");
```
Comment (csperl) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Even though "comment-input" is only used here and in other files, it is still a good practice to define these constants and do not simply use them in line here.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -28,15 +33,33 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // load comments from database
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    // return list of results
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String content = (String) entity.getProperty("content");
```
Comment (csperl) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: According to the documentation of getProperty (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#getproperty): "The value returned may not be the same type as originally set via setProperty(java.lang.String, java.lang.Object)."

Have you checked that? Should we check that it is a String before we do the typecast?

Comment (tatiana-s) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Should I throw an exception if the object isn't a String and if so, what kind would be best?

Comment (csperl) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Now you are ignoring any other content which is fine for now! Thanks!
In practice I would not throw an exception here, but rather log an error. If we throw an exception, then it is broken for the user. If we log an error, then we would see the issue in our logs and we could fix it, but users still might get "some" content.

**Review by csperl — APPROVED** [review]
Thanks for the changes! Well done!

**Pull request merged**